src/main/scala/ai/privado/passes/YamlLinkerPass.scala

```diff
@@ -0,0 +1,40 @@
+package ai.privado.passes
+
+import ai.privado.languageEngine.java.language.NodeStarters
+import ai.privado.tagger.PrivadoParallelCpgPass
+import io.shiftleft.codepropertygraph.generated.{Cpg, EdgeTypes}
+import io.shiftleft.codepropertygraph.generated.nodes.*
+import io.shiftleft.codepropertygraph.generated.nodes.JavaProperty
+import io.shiftleft.semanticcpg.language.*
+
+class YamlLinkerPass(cpg: Cpg) extends PrivadoParallelCpgPass[JavaProperty](cpg) {
```

Comment:
Why do we need a separate pass for this? We can do this inside the `JavaPropertyLinkerPass` itself.

Reply:
Yes, but I believe javaPropertyLinkerPass is intended for Java's .properties files, which handle a specific use case where we fetch the value using the getProperty call. Yes, we can modify the code to use JavaPropertyLinkerPass, but handling the YAML related functionality in a separate pass would make it loosely coupled.

Reply:
The intended purpose of the `JavaPropertyLinkerPass` is to link properties to their respective use sites. We'll have many passes if we want to do it for other property files as well.

Reply:
You can simply add an `if` condition for YAML properties in the `JavaPropertyLinkerPass`.

Reply:
yes agree, In that case we will need to make the `JavaPropertyLinkerPass` more generic, which will require refactoring. @HiteshMah-Jan should we go with one pass for all `property` related linking?

Reply:
As discussed with @HiteshMah-Jan, we are going with separate pass.